lib/fpm/package.rb

```diff
@@ -249,8 +257,20 @@ def output(path)
                                   "creating #{self.type} packages")
   end # def output
 
+  def temporary_directory(path)
+    if attributes[:source_date_epoch].nil?
+      return Stud::Temporary.directory("package-#{type}-staging")
+    else
+      # Choose a predictable build path; see https://reproducible-builds.org/docs/build-path/
+      root = ENV["TMP"] || ENV["TMPDIR"] || ENV["TEMP"] || "/tmp"
```

Comment:
Ahh, that helps a bit. Are there known casess with your `fpm -s gem -t deb json` example where temporary directories (by file content or file path) are showing up somehow in the resulting .deb?